src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/MinMax_PowerFxV1CompatibilityRules.txt

```diff
@@ -0,0 +1,346 @@
+5
+
+>> Min([1, 2, 3, 4, 5], Value)
+1
+
+>> Max([1, 2, 3], 4)
+4
+
+>> Min([1, 2, 3], 4)
+4
+
+>> Max(Table({Name:"John Doe", DOB:Date(1985,3,5), Thing: "1"},{Name:"Jane Roe", DOB:Date(1986,10,19), Thing: "2"}),Thing)
+2
+
+>> Min(Table({Name:"John Doe", DOB:Date(1985,3,5), Thing: "1"},{Name:"Jane Roe", DOB:Date(1986,10,19), Thing: "2"}),Thing)
+1
+
+// ****** Table + expression, date/time ******
+
+>> Max(Table({Name:"John Doe", DOB:Date(1985,3,5)},{Name:"Jane Roe", DOB:Date(1986,10,19)}),DOB)
+Date(1986,10,19)
+
+>> Min(Table({Name:"John Doe", DOB:Date(1985,3,5)},{Name:"Jane Roe", DOB:Date(1986,10,19)}),DOB)
+Date(1985,3,5)
+
+>> Max(Sequence(10), Date(2000,1,10 + Value))
+Date(2000,1,20)
+
+>> Min(Sequence(10), Date(2000,1,10 + Value))
+Date(2000,1,11)
+
+// ****** Errors, table + expression ******
+
+>> Max([1, 2, 1/0, 4, 5], Value)
+Error({Kind:ErrorKind.Div0})
+
+>> Min([1, 2, 1/0, 4, 5], Value)
+Error({Kind:ErrorKind.Div0})
+
+>> Max(If(Sqrt(-1) < 1, [1,2,3]), Value)
+Error({Kind:ErrorKind.Numeric})
+
+>> Min(If(Sqrt(-1) < 1, [1,2,3]), Value)
+Error({Kind:ErrorKind.Numeric})
+
+// ****** Compiler errors ******
+
+>> Max(1, 2, {a:3})
+Errors: Error 10-15: Expected number. We expect a number at this point in the formula.
+
+>> Min(1, 2, {a:3})
+Errors: Error 10-15: Expected number. We expect a number at this point in the formula.
+
+>> Max({a:1}, {a:2}, {a:3})
+Errors: Error 4-9: Expected number. We expect a number at this point in the formula.|Error 11-16: Expected number. We expect a number at this point in the formula.|Error 18-23: Expected number. We expect a number at this point in the formula.
+
+>> Min({a:1}, {a:2}, {a:3})
+Errors: Error 4-9: Expected number. We expect a number at this point in the formula.|Error 11-16: Expected number. We expect a number at this point in the formula.|Error 18-23: Expected number. We expect a number at this point in the formula.
+
+>> Min([1,2,3], [1])
+Errors: Error 13-16: Expected number. We expect a number at this point in the formula.|Error 0-17: The function 'Min' has some invalid arguments.
+
+>> Max([1,2,3], [1])
+Errors: Error 13-16: Expected number. We expect a number at this point in the formula.|Error 0-17: The function 'Max' has some invalid arguments.
+
+>> Min(Date(1985,3,5), DateTime(2014,11,11,13,50,0), Time(12, 1, 7))
+Date(1899,12,30)
+
+>> Min(DateTime(2014,11,11,13,50,0), Date(1985,3,5), Time(12, 1, 7))
+DateTime(1899,12,30,12,1,7,0)
+
+>> Min(Time(12, 1, 7), Date(1985,3,5), DateTime(2014,11,11,13,50,0))
+Time(0,0,0,0)
+
+>> Min(Date(1985,3,5), DateTime(2014,11,11,13,50,0))
+Date(1985,3,5)
+
+>> Min(DateTime(2014,11,11,13,50,0), Date(1985,3,5))
+DateTime(1985,3,5,0,0,0,0)
+
+>> Max("1",2)
+2
+
+>> Max(1,"2")
+2
+
+>> Min(1/0, Date(2023, 4, 20)) 
+Error({Kind:ErrorKind.Div0})
+
+>> Min(Date(2023, 4, 20), 1/0)
+Error({Kind:ErrorKind.Div0})
+
+>> Min(1, "foo")
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> Min(Today(), 3)
+Date(1900,1,2)
+
+>> Min(Now(), "2023")
+2023
```

Comment:
This should be DateTime(1905,7,15,0,0,0,0) - since the first argument is of type DateTime, this dictates the type of the output

Reply:
// "2023" as a string cannot be coerced to a DateTime (invalid format)
>> Min(Now(), "2023")
Error({Kind:ErrorKind.InvalidArgument})

>>Min(Now(), "2023-01-01")
DateTime(2023,1,1,0,0,0,0)